@xwu The fixit does not seem very informative though (maybe that's a yet another bug!):

```
$ swiftc test/Sanitizers/witness_table_lookup.swift
witness_table_lookup.swift:32:16: error: 'where' clause next to generic parameters is obsoleted
struct A<T : Q where T : Q2> : P {
              ~^~~~~~~~~~~~
                                 where T : Q2
```

--- a/include/swift/AST/DiagnosticsParse.def
+++ b/include/swift/AST/DiagnosticsParse.def
@@ -1421,7 +1421,8 @@ WARNING(swift3_where_inside_brackets,none,
         "'where' clause next to generic parameters is deprecated "
         "and will be removed in the future version of Swift", ())
 ERROR(where_inside_brackets,none,
-        "'where' clause next to generic parameters is obsolete", ())
+        "'where' clause next to generic parameters is obsolete, "
+        "must be written following the declaration's type", ())
 
 //------------------------------------------------------------------------------
 // Conditional compilation parsing diagnostics

